src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -769,122 +920,138 @@ default void setUtf8String(long offset, String str) {
 
 
     /**
-     * Creates a buffer memory segment that models the memory associated with the given {@link Buffer} instance.
+     * Creates a memory segment that models the memory associated with the given {@link Buffer} instance.
      * The segment starts relative to the buffer's position (inclusive) and ends relative to the buffer's limit (exclusive).
      * <p>
-     * If the buffer is {@linkplain ByteBuffer#isReadOnly() read-only}, the resulting segment will also be
-     * {@linkplain ByteBuffer#isReadOnly() read-only}. The memory session associated with this segment can either be the
-     * {@linkplain MemorySession#global() global} memory session, in case the buffer has been created independently,
-     * or some other memory session, in case the buffer has been obtained using {@link #asByteBuffer()}.
+     * If the buffer is {@linkplain Buffer#isReadOnly() read-only}, the resulting segment will also be
+     * {@linkplain ByteBuffer#isReadOnly() read-only}. Moreover, if the buffer is a {@linkplain Buffer#isDirect() direct buffer},
+     * the return segment is a native segment; otherwise the returned memory segment is a heap segment.
      * <p>
-     * The resulting memory segment keeps a reference to the backing buffer, keeping it <em>reachable</em>.
+     * The memory session {@code S} associated with the returned segment is
+     * computed as follows:
+     * <ul>
+     *     <li>if the buffer has been obtained by calling {@link #asByteBuffer()} on a memory segment whose session
+     *     is {@code S'}, then {@code S = S'}; or</li>
+     *     <li>if the buffer is an heap buffer, then {@code S} is the {@linkplain MemorySession#global() global session}; or
+     *     <li>if the buffer is a direct buffer, then {@code S} is an
+     *     {@linkplain MemorySession#openImplicit() implicit session} that keeps the buffer reachable.
+     *     Therefore, the off-heap memory associated with the buffer instance will remain available as long as the
+     *     returned segment is reachable.</li>
+     * </ul>
      *
-     * @param buffer the buffer instance backing the buffer memory segment.
-     * @return a buffer memory segment.
+     * @param buffer the buffer instance to be wrapped by a new memory segment.
+     * @return a memory segment, wrapping the given buffer instance.
```

Comment:
Since there's not always a wrapping happening, maybe:
```suggestion
     * @param buffer the buffer instance to be turned into a new memory segment.
     * @return a memory segment, derived from the given buffer instance.
```

Reply:
Good suggestion, we've been generally trying to avoid "wrapping"